### PR TITLE
opendoas: new package at 6.6

### PR DESCRIPTION
**Pull request by ioraff**
## Description of package
Much simpler alternative to `sudo`. Compatibility provided in simultaneous PR to [kiss](https://github.com/kisslinux/kiss) and provided `doas.conf`.

## Checklist

- [x] Latest upstream version.
- [x] I agree to maintain this package (**required**).
- [x] I agree to notify KISS if I can no longer maintain this package (**required**).

### CODE REVIEW

File: community/opendoas/build [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh -e
+
+printf "/chown \${BINOWN}:\${BINGRP}/d\n/chmod \${BINMODE}/d\nw\nq" |
+    ed bsd.prog.mk >/dev/null
```
Comment (dylanaraps) [inline community/opendoas/build]: Rationale behind this?

File: community/opendoas/build [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+./configure \
+    --enable-static \
+    --without-pam
```
Comment (dylanaraps) [inline community/opendoas/build]: What functionality is lost when `pam` is disabled (if any)?

File: community/opendoas/files/doas.conf [hunk 3]
```diff
@@ -0,0 +1,6 @@
+### Select one of the following. Deleting these may cause package installation
+### and removal to function unexpectedly
+
+permit setenv { KISS_FORCE } :wheel cmd kiss
+#permit persist setenv { KISS_FORCE } :wheel cmd kiss
+#permit nopass setenv { KISS_FORCE } :wheel cmd kiss
```
Comment (dylanaraps) [inline community/opendoas/files/doas.conf]: Explanation of these three lines?

Also note: The package manager supports other environment variables such as `KISS_PATH`, `KISS_PID`, etc.

Comment (ioraff) [reply community/opendoas/files/doas.conf]: permits kiss to be run as root if the user belongs to `wheel`, maintaining KISS_FORCE from the invoking user's environment. `persist` caches the privilege escalation for some indeterminate amount of time, and `nopass` allows it to be run without a password. I just commented these out in the same vein as e.g. /etc/sudoers.

Regarding environmental variables, `KISS_FORCE` seems to be the only one used when privileges are escalated (under installation and removal), hence KISS_FORCE="$KISS_FORCE" on lines 969 and 971, with the other environmental variables used only for unprivileged actions such building, etc., but please correct me if I'm wrong.

File: community/opendoas/build (hunk 2 above)
Comment (ioraff) [reply community/opendoas/build]: nothing that matters in a KISS system. `shadow` is still enabled by default.

File: community/opendoas/files/doas.conf (hunk 3 above)
Comment (dylanaraps) [reply community/opendoas/files/doas.conf]: :+1:

File: community/opendoas/build (hunk 1 above)
Comment (ioraff) [reply community/opendoas/build]: no real reason, pretty much just copied over from void, with execute-only permissions assigned in `post-install`.

File: community/opendoas/files/doas.conf (hunk 3 above)
Comment (ioraff) [reply community/opendoas/files/doas.conf]: I've just learned that `persist` requires an additional `configure` option due to it being an unstable feature on Linux, so I've removed that commented-out line

File: community/opendoas/build (hunk 1 above)
Comment (dylanaraps) [reply community/opendoas/build]: Would this work without the `post-install` or this `ed` call?

Comment (ioraff) [reply community/opendoas/build]: yes

Comment (dylanaraps) [reply community/opendoas/build]: So, they could both be removed?

Comment (ioraff) [reply community/opendoas/build]: ... done

File: community/opendoas/files/doas.conf [hunk 4]
```diff
@@ -0,0 +1,5 @@
+### Select one of the following. Deleting these may cause package installation
+### and removal to function unexpectedly
+
+permit setenv { KISS_FORCE } :wheel cmd kiss
+#permit nopass setenv { KISS_FORCE } :wheel cmd kiss
```
Comment (dylanaraps) [inline community/opendoas/files/doas.conf]: No more `persist`?

Comment (ioraff) [reply community/opendoas/files/doas.conf]: I've just learned that it's potentially dangerous on Linux as it requires a different implementation than OpenBSD

Comment (dylanaraps) [reply community/opendoas/files/doas.conf]: Understood. :+1:

**Comment by dylanaraps** [comment]
Thanks :+1:

**Pull request merged**